addons/barcodes/models/barcode_nomenclature.py

```diff
@@ -139,87 +128,32 @@ def parse_barcode(self, barcode):
             'value': 0,
         }
 
-        rules = []
         for rule in self.rule_ids:
-            rules.append({'type': rule.type, 'encoding': rule.encoding, 'sequence': rule.sequence, 'pattern': rule.pattern, 'alias': rule.alias})
-
-        for rule in rules:
             cur_barcode = barcode
-            if rule['encoding'] == 'ean13' and self.check_encoding(barcode,'upca') and self.upc_ean_conv in ['upc2ean','always']:
+            if rule.encoding == 'ean13' and self.check_encoding(barcode,'upca') and self.upc_ean_conv in ['upc2ean','always']:
                 cur_barcode = '0'+cur_barcode
-            elif rule['encoding'] == 'upca' and self.check_encoding(barcode,'ean13') and barcode[0] == '0' and self.upc_ean_conv in ['ean2upc','always']:
+            elif rule.encoding == 'upca' and self.check_encoding(barcode,'ean13') and barcode[0] == '0' and self.upc_ean_conv in ['ean2upc','always']:
                 cur_barcode = cur_barcode[1:]
 
-            if not self.check_encoding(barcode,rule['encoding']):
+            if not self.check_encoding(barcode,rule.encoding):
                 continue
 
-            match = self.match_pattern(cur_barcode, rule['pattern'])
+            match = self.match_pattern(cur_barcode, rule.pattern)
             if match['match']:
-                if rule['type'] == 'alias':
-                    barcode = rule['alias']
+                if rule.type == 'alias':
+                    barcode = rule.alias
                     parsed_result['code'] = barcode
                 else:
-                    parsed_result['encoding'] = rule['encoding']
-                    parsed_result['type'] = rule['type']
+                    parsed_result['encoding'] = rule.encoding
+                    parsed_result['type'] = rule.type
                     parsed_result['value'] = match['value']
                     parsed_result['code'] = cur_barcode
-                    if rule['encoding'] == "ean13":
+                    if rule.encoding == "ean13":
                         parsed_result['base_code'] = self.sanitize_ean(match['base_code'])
-                    elif rule['encoding'] == "upca":
+                    elif rule.encoding == "upca":
```

Comment:
If you don't mind please don't make it so many operation if that same possible in a single operation.
Learn which operation occupies how much time then put it..
Its totally blind migration.
```
parsed_result = dict(
parsed_result,
encoding=rule.encoding,
type=rule.type,
value=match['value']
code=cur_barcode
)
```
:turtle: 
sorry you are doing like this.
![you-can-do-your-job-your-eyes-closed](https://cloud.githubusercontent.com/assets/14940179/12079846/37eac664-b205-11e5-8003-85d5be561986.gif)